--- a/src/ck-sysdeps-linux.c	2017-06-02 03:31:44.000000000 +0200
+++ b/src/ck-sysdeps-linux.c	2018-03-09 17:42:57.270749854 +0100
@@ -561,7 +561,7 @@
                                    &length,
                                    &error);
         if (! res) {
-                g_warning ("Couldn't read %s: %s", path, error->message);
+                g_debug ("Couldn't read %s: %s", path, error->message);
                 g_error_free (error);
                 goto out;
         }
@@ -665,13 +665,13 @@
                                    &length,
                                    &error);
         if (! res) {
-                g_warning ("Couldn't read %s: %s", path, error->message);
+                g_debug ("Couldn't read %s: %s", path, error->message);
                 g_error_free (error);
                 goto out;
         }
 
         if (contents[0] == '\0') {
-                g_warning ("Couldn't read %s: empty file", path);
+                g_debug ("Couldn't read %s: empty file", path);
                 goto out;
         }
 
@@ -683,7 +683,7 @@
         }
 
         if (errno == ERANGE) {
-                g_warning ("Couldn't read %s: %s", path, g_strerror (errno));
+                g_debug ("Couldn't read %s: %s", path, g_strerror (errno));
                 goto out;
         }
 
@@ -819,7 +819,7 @@
         /* get memory data */
         ret = g_file_get_contents (filename, &contents, NULL, &error);
         if (!ret) {
-                g_warning ("failed to open %s: %s", filename, error->message);
+                g_debug ("failed to open %s: %s", filename, error->message);
                 g_error_free (error);
                 goto out;
         }
@@ -889,7 +889,7 @@
         g_debug ("excuting command: %s", command);
         ret = g_spawn_command_line_sync (command, NULL, NULL, &exit_status, &error);
         if (!ret) {
-                g_warning ("failed to run script: %s", error->message);
+                g_debug ("failed to run script: %s", error->message);
                 g_error_free (error);
                 goto out;
         }

